# Assembly and Annotation of Fern Transcriptomes

## 1. Download Sequence Data 

Download reads for transcriptomes generated by [Shen et al. (2018)](https://academic.oup.com/gigascience/article/7/2/gix116/4656250) and [Qi et al. (2018)](https://www.sciencedirect.com/science/article/pii/S1055790318301854?via%3Dihub).  
```
fasterq-dump SRRXXXXXXX --split-files 
```

## 2. Quality Check and Cleaning 

Check the quality of raw reads using [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and trim adaptors and poor-quality reads with [Trimmomatic v. 0.39](http://www.usadellab.org/cms/?page=trimmomatic).
Note that UFRC provides wrapper scripts for these Java programs. 
```
fastqc *.fastq
trimmomatic PE SRRXXXXXXX_1.fastq SRRXXXXXXX_2.fastq SRRXXXXXXX_1P.fastq SRRXXXXXXX_1U.fastq \
SRRXXXXXXX_2P.fastq SRRXXXXXXX_2U.fastq ILLUMINACLIP:TruSeq2-PE.fa:2:30:10 ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 HEADCROP:5
```

## 3. Transcriptome Assembly 

Cleaned reads were assembled following the [One Thousand Plant Transcriptomes Initiative (2019)](https://www.nature.com/articles/s41586-019-1693-2). 
Trimmed reads were assembled using the default parameters of [SOAPdenovo-Trans v.1.03](https://github.com/aquaskyline/SOAPdenovo-Trans)
with a kmer size of 25. [GapCloser v.1.12](https://sourceforge.net/projects/soapdenovo2/files/GapCloser/)
was used following 1KP (2019). 

SOAPdenovo-Trans Config file: 
```
# maximal read length 
max_rd_len=150
[LIB]
#maximal read length in this lib
rd_len_cutoff=150

# average insert size
avg_ins=200

# Insert size < 500, use reverse_seq =0
reverse_seq=0

#in which part(s) of the reads are used 
asm_flags=3

# minumum algined length to contigs for a reliable read location 
map_len=32

# fastq file for read 1
q1=/PATH/SRRXXXXXXX_1P.fastq

# fastq file for read 2
q2=/PATH/SRRXXXXXXX_2P.fastq

```

Reads were used in all steps of the SOAPdenovo-Trans and GapCloser pipeline. 

```
SOAPdenovo-Trans-31mer all -s Nephrolepis.config -o Nephrolepis_out -K 25 -p 32
GapCloser -b Nephrolepis.config -a Nephrolepis_out.scafSeq -o Nephrolepis_gapcloser_out -l 125 -t 32
```

Transcriptome assemblies generated from the One Thousand Plant Transcriptomes Initiative (2019)
were downloaded from [Cyverse](https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/oneKP_capstone_2019). 

## 4. Remove Plastid Contigs

Transcriptomes were Blasted against a database of fern plastome sequences from NCBI, organized by Paul Wolf and 
accessible [here](https://paulwolflab.com/data-protocols/fern_plastome_list/). The plastome database was accessed July 2020. A custom Python script was used to 
remove scaffolds or contigs with significant hits (e-value < 1e-4, overlap >300bp, bitscore > 50). 
```
blastn -db all_plastomes.fasta -query transcriptome.fasta -out transcriptome_plastome_blast \ 
-evalue 0.0001 -outfmt 6 -max_target_seqs 10 -num_threads 16
python filter_contaminants.py transcriptome.fasta transcriptome_plastome_blast > transcriptome_nocp.fasta
```

## 5. Remove Redundant Transcripts 

Transcripts with 98% similarity were collapsed using [CD-HIT v.4.8.6](http://weizhongli-lab.org/cd-hit/).  
```
cd-hit-est -i transcriptome_no_cp.fasta -c 0.98 -o transcriptome_no_cp_cd_hit
```

## 6. Assess Assembly Completeness 

[BUSCO v. 3.02](https://busco.ezlab.org/) was used to assess the completeness of the transcriptome assemblies. The BUSCO databases are not ideal benchmarks for fern lineages as 
fern genomes were not included when developing the ortholog databases. We decided to use the Eukaryote odb9 ortholog set for this reason. 

```
run_BUSCO.py -f --cpu 4 -i transcriptome.fasta -o transcriptome_euk_OUT -l ${HPC_BUSCO_DAT}/eukaryota_odb9 -m tran
```

## 7. Transcriptome Annotation

Peptide and coding sequence files were generated with [TransDecoder v. 5.5](https://github.com/TransDecoder/TransDecoder/wiki).
```
TransDecoder.LongOrfs -t transcriptome_nocp_cdhit.fa
TransDecoder.Predict -t transcriptome_nocp_cdhit.fa
```

[Trinotate v. 3.2](https://github.com/Trinotate/Trinotate.github.io/wiki) was used to annotate each transcriptome. 

``` 
module load ncbi_blast

prot_db="/ufrc/data/reference/trinotate/v4/uniprot_sprot.pep" 

blastx -query transcriptome_nocp_cdhit.fa -db $prot_db -num_threads 80 \
-max_target_seqs 1 -outfmt 6 -evalue 1e-3 > transcriptome_nt.blast

blastp -query transcriptome_nocp_cdhit.transdecoder.pep -db $prot_db \
-num_threads 80 -max_target_seqs 1 -outfmt 6 -evalue 1e-3 > transcriptome_pep.blast

module load hmmer
hmmr_db="/ufrc/data/reference/trinotate/v4/Pfam-A.hmm"
hmmscan --cpu 80 --domtblout transcriptome_TrinotatePFAM.out $hmmr_db \
transcriptome_nocp_cdhit.transdecoder.pep > transcriptome_pfam.log
 
module load signalp/4.1
signalp -f short -n transcriptome_signalp.out transcriptome_nocp_cdhit.transdecoder.pep 

module load trinotate

tmhmm --short < transcriptome_nocp_cdhit.transdecoder.pep  > transcriptome_tmhmm.out

RnammerTranscriptome.pl --transcriptome transcriptome_nocp_cdhit.fa

#use sed to generate two files needed for Trinnoate--
#the gene to transcript and transcript files (note that these commands differ for the 1KP and Qi/Shen et al. assemblies)  
sed 's|\(scaffold[-A-Za-z0-9\_\.]*\)[[:space:]].*[0-9].*\(scaffold[-A-Za-z0-9\.\_]*\);G.*$|\2\t\1|g' \
transcriptome_nocp_cdhit.transdecoder.bed > transcriptome-gene-to-transcript

sed 's|\(scaffold[-A-Za-z0-9\_\.]*\)[[:space:]].*[0-9].*\(scaffold[-A-Za-z0-9\.\_]*\);G.*$|\1|g' \
transcriptome_nocp_cdhit.transdecoder.bed > transcriptome-transcripts.txt

#run custom python script to get sequences that correspond to transcripts from transcriptome fasta
#name output as txm_matching.fasta 
module load python
python extract_seqs_prac_1.py transcriptome-transcripts.txt transcriptome_nocp_cdhit.fa

#download sqlite template database 
wget https://data.broadinstitute.org/Trinity/Trinotate_v3_RESOURCES/Trinotate_v3_RESOURCES/Trinotate_v3.sqlite.gz
gunzip Trinotate_v3.sqlite.gz

Trinotate Trinotate_v3.sqlite init --gene_trans_map transcriptome-gene-to-transcript \
--transcript_fasta transcriptome_nocp_cdhit.fa_matching.fasta \
--transdecoder_pep transcriptome_nocp_cdhit.transdecoder.pep
Trinotate Trinotate_v3.sqlite LOAD_swissprot_blastp transcriptome_pep.blast 
Trinotate Trinotate_v3.sqlite LOAD_swissprot_blastx transcriptome_nt.blast
Trinotate Trinotate_v3.sqlite LOAD_pfam transcriptome_TrinotatePFAM.out 
Trinotate Trinotate_v3.sqlite LOAD_tmhmm transcriptome_tmhmm.out 
Trinotate Trinotate_v3.sqlite LOAD_signalp transcriptome_signalp.out
Trinotate Trinotate_v3.sqlite report > transcriptome_trinotate_annotation_report.xls
```

## References: 

Bryant, D.M., K. Johnson, T. DiTommaso, T. Tickle, M.B. Couger, D. Payzin-Dogru, T.J. Lee, N.D. Leigh, T-H. Kuo, F.G. Davis, J. Bateman,
S. Bryant, A.R. Guzikowski, S.L. Tsai, S. Coyne, W. Ye, R.M. Freedman, L. Peshkin, C.J. Tabin, A. Regev, B.J. Haas, J.L. Whited (2017) 
[A tissue-mapped axolotl de novo transcriptome enables identification of limb regeneration factors.](https://www.cell.com/cell-reports/fulltext/S2211-1247(16)31770-3?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS2211124716317703%3Fshowall%3Dtrue)
<b>18(3):</b> 762-776. 

Fu, L., B. Niu, Z. Zhu, S. Wu, W. Li (2012) 
[CD-HIT: accelerated for clustering the next-generation sequencing data.](https://academic.oup.com/bioinformatics/article/28/23/3150/192160) Bioinformatics <b>28(23):</b> 3150-3152.

Haas, B.J., A. Papnicolau, M. Yassour, M. Grabherr, P.D. Blood, J. Bowden, M.B. Couger, D. Eccles, B. Li, M. Lieber, M.D. MacManes, 
M. Ott, J. Orvis, N. Pouchet, F. Strozzi, N. Weeks, R. Westerman, T. William, C.N. Dewey, R. Henschel, R.D. LeDuc, N. Friedman, A. Regev (2013) 
[De novo transcript sequence reconstruction from RNA-Seq: reference generation and analysis with Trinity.](https://www.nature.com/articles/nprot.2013.084) <b>8:</b> 1494-1512.  

One Thousand Plant Transcriptomes Initiative (2019) 
[One thousand plant transcriptomes and the phylogenomics of green plants.](https://www.nature.com/articles/s41586-019-1693-2) Nature <b>574</b>:679-685. 

Shen, H., D. Jin, J-P. Shu, X-L. Zhu, M. Lei, R. Wei, H. Shang, H-J. Wei, R. Zhang, L. Liu, Y-F. Gu, X-C. Zhang, Y-H. Yan (2018) 
[Large-scale phylogenomic analysis resolves a backbone phylogeny in ferns.](https://academic.oup.com/gigascience/article/7/2/gix116/4656250) Gigascience <b>7</b>: 1-11. 

Simao, F.A., R.M. Waterhouse, P. Ioannidis, E.V. Kriventseva, E.M. Zdobnov (2015) 
[BUSCO: assessing genome assembly and annotation complteness with single-copy orthologs.](https://academic.oup.com/bioinformatics/article/31/19/3210/211866). Bioinformatics <b>31(19):</b> 3210-3212.  

Qi, X., L-Y. Kuo, C. Guo, H. Li, Z. Li, J. Qi, L. Wang, Y. Hu, J. Xiang, C. Zhang, J. Guo, C-H. Huang, H. Ma (2018)
[A well-resolved fern nuclear phylogeny reveals the evolution history of numerous transcription factor families](https://www.sciencedirect.com/science/article/pii/S1055790318301854?via%3Dihub). Molecular Phylogenetics and Evolution <b>127</b>: 961-977. 

Xi, Y., G. Wu, J. Tang, R. Luo, J. Patterson, S. Liu, W. Huang, G. He, S. Gu, S. Li, X. Zhou, T-W. Lam, Y. Li, X. Xu, G. K-S. Wong, J. Wang (2014) 
[SOAPdenovo-Trans: de novo transcriptome assembly with short RNA-Seq reads](https://academic.oup.com/bioinformatics/article/30/12/1660/380938) Bioinformatics <b>30(12):</b> 1660-1666.  




